### PR TITLE
[DOC] Enhanced RDoc for copy_entry

### PR DESCRIPTION
**Pull request by BurdetteLamar**
This PR experiments with using ```tree``` to display directory structure (both src and dest).  If we decide to keep this, we'll need to add a note about tree in the intro.

### CODE REVIEW

File: lib/fileutils.rb [hunk 1]
```diff
@@ -777,21 +777,41 @@ def cp_r(src, dest, preserve: nil, noop: nil, verbose: nil,
   end
   module_function :cp_r
 
+  # Recursively copies files from +src+ to +dest+.
   #
-  # Copies a file system entry +src+ to +dest+.
-  # If +src+ is a directory, this method copies its contents recursively.
-  # This method preserves file types, c.f. symlink, directory...
```
Comment (jeremyevans) [inline lib/fileutils.rb:783]: I don't think we want to drop this information.  If removing this text, we should mention elsewhere what is not supported.

**Review by jeremyevans — CHANGES_REQUESTED** [review]

File: lib/fileutils.rb [hunk 2]
```diff
@@ -777,21 +777,41 @@ def cp_r(src, dest, preserve: nil, noop: nil, verbose: nil,
...
-  # If +preserve+ is true, this method preserves owner, group, and
-  # modified time.  Permissions are copied regardless +preserve+.
+  #   FileUtils.touch('src0.txt')
+  #   File.exist?('dest0.txt') # => false
+  #   FileUtils.copy_entry('src0.txt', 'dest0.txt')
+  #   File.file?('dest0.txt')  # => true
+  #
+  # If +src+ is a directory, recursively copies +src+ to +dest+:
+  #
+  #   system('tree src1')
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply lib/fileutils.rb:791]: That would be fine with me.  Do you think we should use tree?

[the remark this answers is not in the document]
Comment (jeremyevans) [reply lib/fileutils.rb:791]: It does make it a little easier to see what is going on, so I don't have a problem using it here and in the documentation for other recursive copy methods.

File: lib/fileutils.rb (hunk 1 above)
Comment (BurdetteLamar) [reply lib/fileutils.rb:783]: Fixed.

File: lib/fileutils.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply lib/fileutils.rb]: > It does make it a little easier to see what is going on, so I don't have a problem using it here and in the documentation for other recursive copy methods.

Done.

**Comment by BurdetteLamar** [comment]
I've omitted the ```system(tree ...)```  as being unnecessary.

**Review by jeremyevans — APPROVED** [review]

**Pull request merged**